### DISCUSSION TITLE
On loading overlap matrices

### DISCUSSION THREAD

**Discussion by cescorihuela** [zerothi/sisl#806] [Q&A] [answered]
Hello,

I am trying to use a model for graphene that includes two nearest neighbors and a non-identity overlap matrix. As I read from the docs, Overlap class allows to create an overlap object but Hamiltonian object has already its own definition of the overlap matrix. Given a set of parameters for the hamiltonian [r, t], and a set of parameters for the overlap matrix [r2, s], is there a way using the construct() function to load simple non-orthogonal models? I have tried things along the lines:
```python
O = Overlap(geometry)
O.construct([r2, s])
Ham = Hamiltonian(geometry, orthogonal=False, S=O)
Ham.construct([r, t])
```
Also
```python
Ham = Hamiltonian(geometry, orthogonal=False)
Ham.H.construct([r, t])
Ham.S.construct([r2, s])
```

Both methods lead to no success. If you could point a direction, it would help a great deal!

Best,

**Comment by tfrederiksen** [comment, marked as answer]
To build a non-orthogonal hamiltonian for graphene, say with hopping $t$ and overlap $s$ between first neighbours, you could do:
```python
g = sisl.geom.graphene()
H = Hamiltonian(g, orthogonal=False)
r = (0.1, 1.6)  # radii
h1 = (0, 1)  # onsite and overlap for inner sphere
h2 = (t, s)  # hopping and overlap for first shell
H.construct([r, (h1, h2)])
```

**Reply by cescorihuela** [reply]
It works!!! Thank you very much, I didn't think of that!

By the way, big thanks for the amazing code you put forward here. It's really, really nice to work with!

Cheers!

**Reply by zerothi** [reply]
Thanks, your continued use and comments on how to improve will be much appreciated!
